src/workspace-data/data-table/shared/RenameColumnModal.ts

```diff
@@ -0,0 +1,116 @@
+import { Fragment, ReactNode, useState } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common';
+import { ValidatedInput } from 'src/components/input';
+import Modal from 'src/components/Modal';
+import { Ajax } from 'src/libs/ajax';
+import { reportError } from 'src/libs/error';
+import { FormLabel } from 'src/libs/forms';
+import * as Utils from 'src/libs/utils';
+import { isGoogleWorkspace } from 'src/libs/workspace-utils';
+import validate from 'validate.js';
+
+export type RenameColumnModalProps = {
+  onDismiss: () => void;
+  onSuccess: () => void;
+  workspace: any;
+  entityType: string;
+  attributeNames: string[];
+  oldAttributeName: string;
+  dataProvider: any;
+};
+
+export const RenameColumnModal = (props: RenameColumnModalProps): ReactNode => {
+  // State
+  const [newAttributeName, setNewAttributeName] = useState('');
+  const [isBusy, setIsBusy] = useState(false);
+  const { onDismiss, onSuccess, workspace, entityType, attributeNames, oldAttributeName, dataProvider } = props;
+
+  // TODO is there a difference between GCP and azure?
+  //  Composed of only letters, numbers, underscores, or dashes; regex match "[A-z0-9_-]+"
+  // Not one of these reserved words:
+  // “name”
+  // “entityType”
+  // “${entityType}_id”, where ${entityType} is the name of the data table
+  const columnNameErrors = validate.single(newAttributeName, {
+    presence: {
+      allowEmpty: false,
+      message: 'Column name is required',
+    },
+    format: {
+      pattern: `^(?!name$|entityType$|${entityType}_id$)[A-Za-z0-9_-]+$`,
+      flags: 'i',
+      message: Utils.cond(
+        [
+          ['name', 'entityType', `${entityType}_id`].includes(newAttributeName),
+          () =>
+            'Column name cannot be "name", "entityType" or [entityType]_id where [entityType] is the name of the table.',
```

Comment:
Since we know the name of the table, can we use it in this error message?

```suggestion
            `Column name cannot be "name", "entityType" or "${entityType}_id".`,
```